Add documentation for dsc when using POCO's (#111)
MicrosoftDocs/OData-docs: Odata-docs/client/v6/basic-crud-operations.md

**Applies To**: [!INCLUDE[appliesto-odataclient](../../includes/appliesto-odataclient-v6.md)]

> ## With Codegen/Connected service

Reviewer: ```suggestion
## With Connected Service
```
Also this heading should be at different level than "Request an entity set" or "Request an individual entity"

Reviewer: Considering you are introducing a new way of doing things you need a better intro for entire "Basic Crud Operations". It should clearly indicate that there are 2 ways: using ConnService and using POCO.
Are there any benefits or risks for choosing one approach over the other? This should be clearly outlined
In which use cases should one be chosen over the other? (Should also be outlined)

The odata client can be used by generating code using the odata connected service or odata codegen. You can learn more on this  from this [tutorial](https://docs.microsoft.com/en-us/aspnet/web-api/overview/odata-support-in-aspnet-web-api/odata-v4/create-an-odata-v4-client-app "Create an OData v4 Client App (C#)") 

> The code

Reviewer: This seems out of place. You could have a header for "Examples" instead

var people = context.People.Execute();
```

> The `Execute()` API call will return an `IEnumerable<Person>`. To use the async api the call for `Execute` can be replaced with `ExecuteAsync` which returns a `Task<IEnumerable<Person>>`

Reviewer: Instead of discussing the returned type in the docs it might be easier to just specify the type in the code snippet instead of using `var`

Reviewer: Also if you could fix the following lines:
L60, L76: `UserName` --> `userName`
L67: Please be noted --> Please note

context.SaveChanges(); // send the request
```

## Using with POCO's

> While developing you may have the need to use the client without using codegen so as to reduce duplication of code for the models. In this case one can still use the odata client 

Reviewer: ```suggestion
While developing, you may want to use OData Client without a code generation tool like _ODataConnectedService_. In this case, one can still use the OData Client
```

> to access their service.

Reviewer: this line is dangling

In order to do this one can simply extend the `DataServiceContext` class in their code and add some convenience methods as shown below

```csharp
// Person class with Odata annotations
[Key('Id')]
public class Person
{
        public string Id { get; set; }
        public string UserName { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
}

// Dataservice context
 public class Container : DataServiceContext
{
    public Container(Uri serviceRoot) : base(serviceRoot)
    {
              this.People = base.CreateQuery<Person>("People");
    }

>     public DataServiceQuery<Person> People {get; }

Reviewer: ```suggestion
    public DataServiceQuery<Person> People { get; }
```

    public void AddToPeople(Person person)
    {

>        base.AddObject("People",person);

Reviewer: ```suggestion
       base.AddObject("People", person);
```

    }
}

```

The above context can then be used the same way as the codegen data service context. 

> To add new objects to the peoples entity set use `context.AddObject("People",person)` or create a convenience method in the class to do the same.

Reviewer: ```suggestion
To add new objects to the peoples entity set use `context.AddObject("People", person)` or create a convenience method in the class to do the same.
```

### Loading the service model

This example allows you to use the client while using the current metadata which is fetched from network. In some cases this is not convenient and to override this behavior you need to change the constructor of the DataService context to include an action that can be used to fetch the model. Codegen tools do this by downloading the metadata and storing it in a verbatim string or a separate file which is then read upon instantiation of the context.

To provide the same for our code we need to do the following.

```csharp
    // ctor from above
    public Container(Uri serviceRoot) : base(serviceRoot)
    {
        this.People = base.CreateQuery<Person>("People");
        // add the following lines

>         this.Format.LoadServiceModel= ()=> util.GetEdmModel()  /* user action that returns a valid IEdmModel instance */  

Reviewer: ```suggestion
        this.Format.LoadServiceModel = () => util.GetEdmModel()  /* user action that returns a valid IEdmModel instance */  
```
